app/assets/javascripts/ransack_ui_jquery/index.js

```diff
@@ -1,2 +1,6 @@
 //= require ransack/predicates
-//= require ransack_ui_jquery/search_form
\ No newline at end of file
+//= require ransack_ui_jquery/search_form
+
+if(jQuery.fn.select2) {
+  window.Select2 = jQuery.fn.select2;
+}
```

Comment:
>  Originally we we're checking for (window) Select2, hence the revert

Ah ok. I thought the idea with the aliasing was to avoid needing the revert (hence "forward compatible")... Cristian makes a good point though, it might be all-or-nothing anyway so a full revert might be necessary.